samza-core/src/main/java/org/apache/samza/standalone/StandaloneJobCoordinator.java

```diff
@@ -62,87 +61,78 @@
-
-    /** TODO:
-     * Locality Manager seems to be required in JC for reading locality info and grouping tasks intelligently and also,
-     * in SamzaContainer for writing locality info to the coordinator stream. This closely couples together
-     * TaskNameGrouper with the LocalityManager! Hence, groupers should be a property of the jobcoordinator
-     * (job.coordinator.task.grouper, instead of task.systemstreampartition.grouper)
-     */
-    this.jobModel = JobModelManager.readJobModel(this.config, Collections.emptyMap(), null, streamMetadataCache, null);
+    this.config = config;
   }
 
   @Override
   public void start() {
     // No-op
-    JobModel jobModel = getJobModel();
-    containerController.startContainer(
-        jobModel.getContainers().get(getProcessorId()),
-        jobModel.getConfig(),
-        jobModel.maxChangeLogStreamPartitions);
+    JobModel jobModel = null;
+    try {
+      jobModel = getJobModel();
+    } catch (Exception e) {
+      LOGGER.error("Exception while trying to getJobModel.", e);
+      if (coordinatorListener != null) {
+        coordinatorListener.onCoordinatorFailure(e);
+      }
+    }
+    if (jobModel != null && jobModel.getContainers().containsKey(processorId)) {
+      if (coordinatorListener != null) {
+        coordinatorListener.onNewJobModel(processorId, jobModel);
+      }
+    } else {
+      stop();
+    }
   }
 
   @Override
   public void stop() {
     // No-op
-    containerController.shutdown();
+    if (coordinatorListener != null) {
+      coordinatorListener.onJobModelExpired();
+      coordinatorListener.onCoordinatorStop();
+    }
   }
 
-  /**
-   * Waits for a specified amount of time for the JobCoordinator to fully start-up, which means it should be ready to
-   * process messages. In a Standalone use-case, it may be sufficient to wait for the container to start-up. In case of
-   * ZK based Standalone use-case, it also includes registration with ZK, the initialization of leader elector module etc.
-   *
-   * @param timeoutMs Maximum time to wait, in milliseconds
-   */
   @Override
-  public boolean awaitStart(long timeoutMs) throws InterruptedException {
-    return containerController.awaitStart(timeoutMs);
+  public String getProcessorId() {
+    return processorId;
   }
 
   @Override
-  public String getProcessorId() {
-    return processorId;
+  public void setListener(JobCoordinatorListener listener) {
+    this.coordinatorListener = listener;
   }
 
   @Override
   public JobModel getJobModel() {
-    return jobModel;
+    JavaSystemConfig systemConfig = new JavaSystemConfig(this.config);
+    Map<String, SystemAdmin> systemAdmins = new HashMap<>();
+    for (String systemName: systemConfig.getSystemNames()) {
+      String systemFactoryClassName = systemConfig.getSystemFactory(systemName);
+      if (systemFactoryClassName == null) {
+        LOGGER.error(String.format("A stream uses system %s, which is missing from the configuration.", systemName));
```

Comment:
I don't think the exception will get logged if the caller doesn't handle the exception, will it? 
If exception gets logged either ways, then, I can remove the redundant log statement.